service/controller/app/resource/releasemigration/create.go

```diff
@@ -43,6 +44,35 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return nil
 	}
 
+	// Resource is used to migrating all charts. it's too noisy if it works on every reconciliation.
+	if key.AppName(cr) != key.ChartOperatorAppName {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("no need to migrate %#q", key.AppName(cr)))
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+		return nil
+	}
+
+	v, err := semver.NewVersion(key.Version(cr))
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if v.Major() < 1 {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("chart %#q's current version %#q is not greater than 1.0.0.", key.AppName(cr), key.Version(cr)))
```

Comment:
I think the log message needs to explain we don't trigger the migration because the tenant cluster uses Helm 2.

For others outside our team they may not be aware 1.X is Helm 3 and 0.X is Helm 2.